src/iotjs_binding_helper.c

```diff
@@ -142,7 +142,11 @@ int iotjs_process_exitcode() {
 
   jerry_value_t jexitcode =
       iotjs_jval_get_property(process, IOTJS_MAGIC_STRING_EXITCODE);
-  IOTJS_ASSERT(jerry_value_is_number(jexitcode));
+
+  if (!jerry_value_is_number(jexitcode)) {
+    jerry_release_value(jexitcode);
+    jexitcode = jerry_create_number(1);
```

Comment:
I'm not sure this is the correct solution. I've checked the node.js to see what the returned exit code will be in case of the test. In node.js it returned `0`. So I think we should do the same here.

Another Node.js example:
```sh
$ node -e "process.exitCode = {valueOf: function(){return 1; }};"
$ echo $?
1
```

It seems that in node the toInteger operation is applied to the `exitCode` value.
IMHO reasonable thing to adapt in IoT.js, just make the engine more node-compatible.

Reply:
Good opinion, I'll update the commit by your comment.
thank you.